### PR TITLE
Addressing #43 [Preview feature]

### PR DESCRIPTION
**Pull request by YorgunKelebek**
Added preview support - works on the single post view with a query paramater mode=preview
Single post view now uses URL Slug in place of the item codename in order to allow preview directly from Kentico Kontent

Prompts for the Kentico API preview token

### CODE REVIEW

**Comment by YorgunKelebek** [comment]
image

image

File: js/app.js [hunk 1]
```diff
@@ -11,3 +11,9 @@ window.addEventListener('DOMContentLoaded', (event) => {
     loadBlogItems(blogCodename);
     loadAsideBlogTags();
 });
+
+
+export function getUrlParamater(param) {
+    const url = new URL(location.href);
+    return new Set(url.searchParams.getAll(param));
```
Comment (goofballLogic) [inline js/app.js:18]: Interesting. Why use Set here?

File: js/blog-tags.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { buildTaxonomiesUrl, fetchKontent, baseUrl } from "./kontent-fetching.js";
+import { getUrlParamater } from "./app.js";
```
Comment (goofballLogic) [inline js/blog-tags.js:2]: This the best place for this function?

File: js/app.js (hunk 1 above)
Comment (YorgunKelebek) [reply js/app.js:18]: Probably because you recommended at some point..?

File: js/blog-tags.js (hunk 2 above)
Comment (YorgunKelebek) [reply js/blog-tags.js:2]: Probably not, but at the moment we don't have any other generic file, such as utils.js

**Pull request merged**